utils.sh

```diff
@@ -56,6 +63,58 @@ function patch_node_ignition() {
     ssh < "${wd}/${kind}.yaml" "${ssh_opts[@]}" "core@$bootstrap" sudo dd of="$config"
 }
 
+function machineconfig_generate_patches() {
+    set -x
+    local kind
+    local path
+    local value
+    local search_path
+    local config_type
+    local files_template
+    local units_template
+    local octal_mode
+    local decimal_mode
+    local dest_path
+    local ifs
+    local simple_name
+
+    read -r -d '' files_template <<'EOF'
```

Comment:
```suggestion
    files_template=$(cat <<EOF
```

Reply:
without the single quoting of EOF, it will try to substitute the variables, which is not the intended behavior at this point (it is done later with envsubst)